### PR TITLE
Fix crashes & correct viewWillAppear & didAppear

### PR DESCRIPTION
**Pull request by snakecharmer**

### CODE REVIEW

File: Pod/Classes/PagingMenuController.swift [hunk 1]
```diff
@@ -9,484 +9,570 @@
...
+			} else {
+				toAttribute = attribute
+			}
+			
+			NSLayoutConstraint(item: imageView, attribute: attribute, relatedBy: .Equal, toItem: toItem, attribute: toAttribute, multiplier: 1, constant: 0).active = true
+		}
+		
+		return imageView
+	}
+	private lazy var frontImageView: UIImageView = self.viewCopyImageView(isFront: true)
```
Comment (kitasuke) [inline Pod/Classes/PagingMenuController.swift]: What's this variable for?

Comment (snakecharmer) [inline Pod/Classes/PagingMenuController.swift]: It's used to generate a uimageview of the old vc to smooth the transition to the new one (cos in the meantime if the controller refresh the data on viewWillAppear the appearance of the updated view controller can be different)

**Comment by kitasuke** [comment]
Please follow existing indent. It's hard to see diff on Github.

File: Pod/Classes/PagingMenuController.swift [hunk 2]
```diff
@@ -9,484 +9,570 @@
...
-        } else {
-            guard index >= previousIndex && index <= nextIndex else { return false }
-        }
-        return true
-    }
+	
+	public weak var delegate: PagingMenuControllerDelegate?
+	private var options: PagingMenuOptions = PagingMenuOptions()
+	public private(set) var menuView: MenuView!
+	public var currentPage: Int {
```
Comment (kitasuke) [inline Pod/Classes/PagingMenuController.swift]: Can you explain why this variable is not public readonly any more?

File: Pod/Classes/PagingMenuController.swift [hunk 3]
```diff
@@ -9,484 +9,570 @@
...
+		let page = Int(offsetX / view.bounds.width)
+		
+		if case .Infinite = options.menuDisplayMode {
+			return max(0, page - 1)
+		} else {
+			return page
+		}
+	}
+	
+	public var currentViewController: UIViewController? {
```
Comment (kitasuke) [inline Pod/Classes/PagingMenuController.swift]: Can you explain why this variable is not public readonly any more?

**Comment by kitasuke** [comment]
Can you explain why all the view controllers is loaded when PagingMenuController is instantiated?
Only three view controllers(previous, visible and next one) used to be loaded at that time due to performance issue.

**Comment by snakecharmer** [comment]
Ok we'll work to fix that, when ready we'll open another pull-request, merry christmas and a happy new year 2016!

**Comment by kitasuke** [comment]
Looking forward to seeing it :) 
Merry christmas and a happy new year too!

**Comment by kitasuke** [comment]
You can also separate your PR like fixing using unowned to weak. I agree with that and it's not related the issue above.

**Pull request closed without merging**